src/commands/virtualdevices.ts

```diff
@@ -0,0 +1,110 @@
+import { type ArgumentsCamelCase, type Argv, type CommandModule } from 'yargs'
+
+import {
+	type Device,
+	DeviceIntegrationType,
+	type DeviceListOptions,
+} from '@smartthings/core-sdk'
+
+import { withLocationsAndRooms, type WithNamedRoom } from '../lib/api-helpers.js'
+import { buildTableOutput } from '../lib/command/util/devices-table.js'
+import { apiCommand, apiCommandBuilder, type APICommandFlags } from '../lib/command/api-command.js'
+import {
+	outputItemOrList,
+	outputItemOrListBuilder,
+	type OutputItemOrListConfig,
+	type OutputItemOrListFlags,
+} from '../lib/command/listing-io.js'
+
+
+export type CommandArgs = APICommandFlags & OutputItemOrListFlags & {
+	idOrIndex?: string
+	location?: string[] | string
+	installedApp?: string
+	verbose: boolean
+}
+
+const command = 'virtualdevices [id-or-index]'
+
+const describe = 'list all virtual devices available in a user account or retrieve a single device'
+
+const builder = (yargs: Argv): Argv<CommandArgs> =>
+	outputItemOrListBuilder(apiCommandBuilder(yargs))
+		.positional(
+			'id-or-index',
+			{ describe: 'the device id or number from list', type: 'string' },
+		)
+		.option(
+			'location',
+			{ alias: 'l', describe: 'filter results by location', type: 'string', array: true },
+		)
+		.option(
+			'installed-app',
+			{
+				alias: 'a',
+				describe: 'filter results by installed app that created the device',
+				type: 'string',
+			},
+		)
+		.option(
+			'verbose',
+			{
+				alias: 'v',
+				describe: 'include location and room name in output',
+				type: 'boolean',
+				default: false,
+			},
+		)
+		.example([
+			['$0 virtualdevices', 'list all virtualdevices'],
+			[
+				'$0 virtualdevices 1',
+				'display details for the first virtualdevices in the list retrieved by running ' +
```

Comment:
The help text descriptions here are awkward.  I'd prefer it to be rendered as "virtual device" or "virtual devices" rather than the command name.

Reply:
Fixed this and the other two examples.